### PR TITLE
Fix assertions for #139

### PR DESCRIPTION
**Pull request by mllg**
Addresses the issues outlined in #139.

The following assertions were missing or broken:

## Assertion on correct task_type

`Filter$task_type` was documented to take a scalar string, but in practice could have multiple values. Renamed to `task_types` (breaking change) and adjusted documentation. Assertion added to `Filter$calculate()`.

## Assertion on non-missing values

Assertion added to `Filter$calculate()`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,46 +48,99 @@ set.seed(1)
...
+| carsurvscore      | Correlation-Adjusted coRrelation Survival Score          | Surv           | Integer, Numeric                                               | [carSurv](https://cran.r-project.org/package=carSurv) , [mlr3proba](https://cran.r-project.org/package=mlr3proba) | NA        |
+| cmim              | Minimal Conditional Mutual Information Maximization      | Classif & Regr | Integer, Numeric, Factor, Ordered                              | [praznik](https://cran.r-project.org/package=praznik)                                                             | NA        |
+| correlation       | Correlation                                              | Regr           | Integer, Numeric                                               | stats                                                                                                             | NA        |
+| disr              | Double Input Symmetrical Relevance                       | Classif & Regr | Integer, Numeric, Factor, Ordered                              | [praznik](https://cran.r-project.org/package=praznik)                                                             | NA        |
```
Comment (pat-s) [inline README.md:129]: The README update sets a lot of values to NA. Is this caused by the PR or a general issue?

File: README.md [hunk 2]
```diff
@@ -48,46 +48,99 @@ set.seed(1)
 library("mlr3")
 library("mlr3filters")
 
-task = tsk("pima")
+task = tsk("sonar")
```
Comment (pat-s) [inline README.md:51]: Why change the README task?

**Review by pat-s — COMMENTED** [review]
Thanks!

Is this really a breaking change for users?

Most changes are internal and the only user facing one I see is the `task_types` renaming in `Filter`. But this is not something users typically use. The assertions would fall into category "bug fix" for me.

File: README.md (hunk 1 above)
Comment (mllg) [reply README.md]: I've updated the scripts to generate the table.

**Pull request merged**